src/test/scala/ru/tinkoff/gatling/amqp/examples/RabbitMQConsumer.scala

```diff
@@ -0,0 +1,79 @@
+package ru.tinkoff.gatling.amqp.examples
+
+import com.rabbitmq.client._
+
+import java.nio.charset.StandardCharsets
+
+// Testclient which consumes messages and writes something in other queue
+
+object RabbitMQConsumer {
+
+  val deliverCallback: DeliverCallback = {
+    new DeliverCallback {
+      override def handle(consumerTag: String, message: Delivery): Unit = {
+        println("Received a message")
+        val connection = getConnection(writePort)
+        val channel = connection.createChannel()
+        channel.queueDeclare(writeQueue, true, false, false, null)
+        channel.basicPublish("", writeQueue, message.getProperties, "Message processed".getBytes(StandardCharsets.UTF_8))
+      }
+    }
+  }
+
+  val cancelCallback = new CancelCallback {
+    override def handle(consumerTag: String): Unit = {
+    }
+  }
+
+  def readAndWrite() = {
+    val connection = getConnection(readPort)
+    val channel = connection.createChannel()
+    channel.queueDeclare(readQueue, true, false, false, null)
+    channel.basicConsume(readQueue,true, deliverCallback, cancelCallback)
+
+  }
+
+  def main(args: Array[String]): Unit = {
+    setup()
+    readAndWrite()
+    Thread.sleep(1000000)
+    tearDown()
+    System.exit(1)
+  }
+
+  private val readQueue = "readQueue"
+  private val readPort = 5672
+
+  private val writeQueue = "writeQueue"
+  private val writePort = 5673
+
+  def tearDown(): Unit = {
+    val readConnection: Connection = getConnection(readPort)
```

Comment:
Better make global readConnection and writeConnection in upper of object. Analogously for channels except maybe channel in DeliverCallback, because this channel in in consumer thread

Reply:
i made the readChannel and writechannel to global variables. The readConnection was always used in conjunction with the readChannel (same for writeChannel) and so its not need. I am brave enough to say, this has been fixed.